src/Altinn.Platform/Altinn.Platform.Authorization/Authorization/Services/Implementation/ContextHandler.cs

```diff
@@ -18,14 +25,167 @@ namespace Altinn.Platform.Authorization.Services.Implementation
         public async Task<XacmlContextRequest> Enrich(XacmlContextRequest request)
         {
+            await EnrichResourceAttributes(request);
             return await Task.FromResult(request);
         }
+
+        private async Task EnrichResourceAttributes(XacmlContextRequest request)
+        {
+            string orgAttributeValue = string.Empty;
+            string appAttributeValue = string.Empty;
+            string instanceAttributeValue = string.Empty;
+            string resourcePartyAttributeValue = string.Empty;
+            string taskAttributeValue = string.Empty;
+            string userAttributeValue = string.Empty;
+
+            XacmlContextAttributes resourceContextAttributes = request.GetResourceAttributes();
+
+            foreach (XacmlAttribute attribute in resourceContextAttributes.Attributes)
+            {
+                if (attribute.AttributeId.OriginalString.Equals(XacmlRequestAttribute.OrgAttribute))
+                {
+                    orgAttributeValue = attribute.AttributeValues.First().Value;
+                }
+
+                if (attribute.AttributeId.OriginalString.Equals(XacmlRequestAttribute.AppAttribute))
+                {
+                    appAttributeValue = attribute.AttributeValues.First().Value;
+                }
+
+                if (attribute.AttributeId.OriginalString.Equals(XacmlRequestAttribute.InstanceAttribute))
+                {
+                    instanceAttributeValue = attribute.AttributeValues.First().Value;
+                }
+
+                if (attribute.AttributeId.OriginalString.Equals(XacmlRequestAttribute.PartyAttribute))
+                {
+                    resourcePartyAttributeValue = attribute.AttributeValues.First().Value;
+                }
+
+                if (attribute.AttributeId.OriginalString.Equals(XacmlRequestAttribute.TaskAttribute))
+                {
+                    taskAttributeValue = attribute.AttributeValues.First().Value;
+                }               
+            }
+
+            bool resourceAttributeComplete = false;
+
+            if (!string.IsNullOrEmpty(orgAttributeValue) &&
+                !string.IsNullOrEmpty(appAttributeValue) &&
+                !string.IsNullOrEmpty(instanceAttributeValue) &&
+                !string.IsNullOrEmpty(resourcePartyAttributeValue) &&
+                !string.IsNullOrEmpty(taskAttributeValue))
+            {
+                // The resource attributes are complete
+                resourceAttributeComplete = true;
+            }
+            else if (!string.IsNullOrEmpty(orgAttributeValue) &&
+                !string.IsNullOrEmpty(appAttributeValue) &&
+                string.IsNullOrEmpty(instanceAttributeValue) &&
+                !string.IsNullOrEmpty(resourcePartyAttributeValue) &&
+                string.IsNullOrEmpty(taskAttributeValue))
+            {
+                // The resource attributes are complete
+                resourceAttributeComplete = true;
+            }
```

Comment:
If I have understood this correctly, instance&task must both be populated or empty but the remaining must always be populated this can be simplified to a single check as 
```
!string.IsNullOrEmpty(orgAttributeValue) &&
!string.IsNullOrEmpty(appAttributeValue) &&
!string.IsNullOrEmpty(resourcePartyAttributeValue) &&
( string.IsNullOrEmpty(instanceAttributeValue) && string.IsNullOrEmpty(taskAttributeValue))
```

Reply:
No, this single check will not solve the purpose. The resourceattributes will be complete when they are not empty as well.  The instance and task can be null or empty only when the org, app, resourceparty are not null